app/controllers/username_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class UsernameController < ApplicationController
+  before_action :authenticate_user!
+
+  def edit; end
+
+  def update
+    current_user.username = params[:user][:username]
+    current_user.save!
```

Comment:
```
current_user.update!(user_params)
```

together with a private method
```
def user_params
  params.require(:user).permit(:username)
end
```
is how the pattern we tend to use.

Reply:
Oh, also, do we want to validate it? In particular do we care about maximal length? Something to check with Isabelle. 

Then we would add validation to user model, check if the `update` without exclamation mark succeeds, and if not, we would render `edit`.

Reply:
Clarified with Isabelle, no validation for now.